api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1036,6 +1036,12 @@ type ClusterNetworking struct {
 	//
 	// +optional
 	APIServer *APIServerNetworking `json:"apiServer,omitempty"`
+
+	// ipv4 allows users to configure IP settings for IPv4 connections. When ommitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv4OVNKubernetes *IPv4OVNKubernetesConfig `json:"ipv4OVNKubernetes,omitempty"`
```

Comment:
I think we can add a struct like this:
```
OVN *OVNKubernetesConfig `json:"ovn,omitempty"`
type OVNKubernetesConfig struct {
	IPv4 *OVNIPv4Config `json:"ipv4,omitempty"`
}
```
This approach will make it easier for us to add other features in the future.